store/types/validity.go

```diff
@@ -1,15 +1,30 @@
 package types
 
+import math "math"
+
+const (
+	// 32K
+	MaxKeyLength = math.MaxUint16
+	// 2G
+	MaxValueLength = math.MaxUint32
+)
+
 // AssertValidKey checks if the key is valid(key is not nil)
 func AssertValidKey(key []byte) {
 	if len(key) == 0 {
 		panic("key is nil")
 	}
+	if len(key) > MaxKeyLength {
+		panic("key is too large")
+	}
```

Comment:
Constraint on key should be reflected in doc comment

Reply:
good catch, done